src/_pytest/config/__init__.py

```diff
@@ -907,6 +908,10 @@ def _getconftest_pathlist(self, name, path):
             values.append(relroot)
         return values
 
+    if six.PY3:
+        # once we drop Python 2, please change this to use the normal decorator syntax (#4227)
+        _getconftest_pathlist = functools.lru_cache(maxsize=None)(_getconftest_pathlist)
```

Comment:
Old school style! 😎

Reply:
\o/

Reply:
btw

```
if six.PY3:
```
is going to [break when python4 comes out](https://github.com/benjaminp/six/blob/e114efceea962fb143c909c904157ca994246fd2/six.py#L37) -- usually prefer `if six.PY2: ... ` / `if not six.PY2: ...`

I've seen other projects do this in a different way:

```python
if six.PY2:
    def lru_cache(*_, **__):
        def dec(fn):
            return fn
        return dec
else:
    from functools import lru_cache
```

neat trick though (saw this from twitter, missed the PR notification)

Reply:
@asottile 
Please consider creating a follow up PR - otherwise I might do it for the follow up via https://github.com/pytest-dev/pytest/pull/4224 (the removed, 2nd commit).
It should go into `pytest.compat` probably, so that it is easier to use everywhere.

Reply:
All fair points. I don't mind leaving the code as is, and also don't mind to change to the ones suggested here too. 👍

Reply:
Will move it to compat now.